#1128: Warenkorb leeren nach automatischem Logout in der Selbstbedienung

pabneukistl opened the pull request.

pabneukistl commented:
Ich hätte jetzt zu der ersten Umsetzung emptyCart bei Autologout in SB noch eine Funktionalität zum Versenden einer Email an den SB-Kunden eingefügt in welcher man den geleerten Einkauf sieht. Zur Nachvollziehbarkeit, dass man dann schauen welche Produkte im Warenkorb waren und kann ob der Einkauf mitgenommen und nur nicht verbucht wurde.. ich weiß aber nicht ob/wie ich Emailversand testen kann? Geht das am Testsystem? Jetzt weiß ich nicht ob das funktioniert?

mrothauer commented:
@pabneukistl sorry hat ein bissl gedauert

hab dir den develop-branch reingemergt 

Ja, E-Mails kann man abtesten, hier ein Beispiel:
https://github.com/foodcoopshop/foodcoopshop/blob/96f09799bfaf58d1be74b574352d147eab92f941/plugins/Admin/tests/TestCase/src/Controller/OrderDetails/OrderDetailsControllerEditPickupDayTest.php#L50-L55

vor den spezifischen asserts muss im test ` $this->runAndAssertQueue();` aufgerufen werden. der 0er ist der index  - die wie vielete mail) (es könnten ja mehrere verschickt worden sein.

pabneukistl commented:
Hallo Mario! Du wo muss ich einen neuen Test eintragen damit der auch aufgerufen wird? Ich weiß ich habs schon mal wo eingetragen damals bei den "Barcode mit Gewicht" Umsetzungen aber ich finds absolut nimmer :-)
<img width="631" alt="image" src="https://github.com/user-attachments/assets/9ff4c56f-5ef4-406b-a602-abb8a7aa6777" />

mrothauer commented:
Der CartType sollte in Customer::getCartType() korrekt gesetzt sein, sofern du die `getSelfServicePostOptions` verwendest (was du auch machst).

Ich kanns jetzt leider auch nicht debuggen, musst dich Schritt für Schritt vorhangeln, ob du im Test überhaupt CartsController::emptyCart aufrufst. Und auch, ob der autologout-Parameter korrekt übergeben wird Stichwort (POST/GET, QueryParam, Data).

pabneukistl commented:
<html>
<body>
<!--StartFragment--><header style="box-sizing: border-box; flex: 1 1 0%; background-color: rgb(211, 60, 71); color: rgb(255, 255, 255); padding: 30px; font-family: &quot;Helvetica Neue&quot;, Helvetica, Arial, sans-serif; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: 0.14px; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><h1 class="header-title" style="box-sizing: border-box; display: flex; align-items: center; font-size: 30px; margin: 0px;"><span style="box-sizing: border-box;">Database Error</span><a style="box-sizing: border-box; font-size: 18px; cursor: pointer; margin-left: 10px; user-select: none;">📋</a></h1><span class="header-type" style="box-sizing: border-box; display: block; font-size: 16px;">Cake\Database\Exception\QueryException</span></header><div class="error-content" style="box-sizing: border-box; padding: 30px; color: rgb(64, 64, 65); font-family: &quot;Helvetica Neue&quot;, Helvetica, Arial, sans-serif; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: 0.14px; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><p class="error-subheading" style="box-sizing: border-box; border-radius: 5px; font-size: 18px; margin-top: 0px; padding: 20px 16px; color: rgb(255, 255, 255); background-color: rgb(49, 151, 149);"><strong style="box-sizing: border-box; color: rgb(255, 255, 255); background-color: rgb(79, 209, 197); border-radius: 9999px; padding: 4px 12px; margin-right: 8px;">Error</strong><span> </span>SQLSTATE[42S22]: Column not found: 1054 Unknown column 'Products.new' in 'where clause' Query: SELECT DISTINCT Products.id_product AS Products__id_product, Products.id_manufacturer AS Products__id_manufacturer, Products.id_tax AS Products__id_tax, Products.id_storage_location AS Products__id_storage_location, Products.price AS Products__price, Products.name AS Products__name, Products.description AS Products__description, Products.description_short AS Products__description_short, Products.unity AS Products__unity, Products.is_declaration_ok AS Products__is_declaration_ok, Products.is_stock_product AS Products__is_stock_product, Products.active AS Products__active, Products.delivery_rhythm_type AS Products__delivery_rhythm_type, Products.delivery_rhythm_count AS Products__delivery_rhythm_count, Products.delivery_rhythm_first_delivery_day AS Products__delivery_rhythm_first_delivery_day, Products.delivery_rhythm_order_possible_until AS Products__delivery_rhythm_order_possible_until, Products.delivery_rhythm_send_order_list_weekday AS Products__delivery_rhythm_send_order_list_weekday, Products.delivery_rhythm_send_order_list_day AS Products__delivery_rhythm_send_order_list_day, Products.created AS Products__created, Products.modified AS Products__modified, DepositProducts.id AS DepositProducts__id, DepositProducts.id_product AS DepositProducts__id_product, DepositProducts.id_product_attribute AS DepositProducts__id_product_attribute, DepositProducts.deposit AS DepositProducts__deposit, Images.id_image AS Images__id_image, Taxes.id_tax AS Taxes__id_tax, Taxes.rate AS Taxes__rate, Taxes.active AS Taxes__active, Taxes.deleted AS Taxes__deleted, Manufacturers.id_manufacturer AS Manufacturers__id_manufacturer, Manufacturers.name AS Manufacturers__name, Manufacturers.description AS Manufacturers__description, Manufacturers.short_description AS Manufacturers__short_description, Manufacturers.created AS Manufacturers__created, Manufacturers.modified AS Manufacturers__modified, Manufacturers.active AS Manufacturers__active, Manufacturers.is_private AS Manufacturers__is_private, Manufacturers.uid_number AS Manufacturers__uid_number, Manufacturers.additional_text_for_invoice AS Manufacturers__additional_text_for_invoice, Manufacturers.iban AS Manufacturers__iban, Manufacturers.bic AS Manufacturers__bic, Manufacturers.bank_name AS Manufacturers__bank_name, Manufacturers.firmenbuchnummer AS Manufacturers__firmenbuchnummer, Manufacturers.firmengericht AS Manufacturers__firmengericht, Manufacturers.aufsichtsbehoerde AS Manufacturers__aufsichtsbehoerde, Manufacturers.kammer AS Manufacturers__kammer, Manufacturers.homepage AS Manufacturers__homepage, Manufacturers.id_customer AS Manufacturers__id_customer, Manufacturers.variable_member_fee AS Manufacturers__variable_member_fee, Manufacturers.send_invoice AS Manufacturers__send_invoice, Manufacturers.send_order_list AS Manufacturers__send_order_list, Manufacturers.default_tax_id AS Manufacturers__default_tax_id, Manufacturers.default_tax_id_purchase_price AS Manufacturers__default_tax_id_purchase_price, Manufacturers.send_order_list_cc AS Manufacturers__send_order_list_cc, Manufacturers.send_instant_order_notification AS Manufacturers__send_instant_order_notification, Manufacturers.send_ordered_product_deleted_notification AS Manufacturers__send_ordered_product_deleted_notification, Manufacturers.send_ordered_product_price_changed_notification AS Manufacturers__send_ordered_product_price_changed_notification, Manufacturers.send_ordered_product_amount_changed_notification AS Manufacturers__send_ordered_product_amount_changed_notification, Manufacturers.enabled_sync_domains AS Manufacturers__enabled_sync_domains, Manufacturers.stock_management_enabled AS Manufacturers__stock_management_enabled, Manufacturers.send_product_sold_out_limit_reached_for_manufacturer AS Manufacturers__send_product_sold_out_limit_reached_for_manufacturer, Manufacturers.send_product_sold_out_limit_reached_for_contact_person AS Manufacturers__send_product_sold_out_limit_reached_for_contact_person, Manufacturers.no_delivery_days AS Manufacturers__no_delivery_days, Manufacturers.include_stock_products_in_order_lists AS Manufacturers__include_stock_products_in_order_lists, Manufacturers.send_delivery_notes AS Manufacturers__send_delivery_notes, Manufacturers.anonymize_customers AS Manufacturers__anonymize_customers, UnitProducts.id AS UnitProducts__id, UnitProducts.id_product AS UnitProducts__id_product, UnitProducts.id_product_attribute AS UnitProducts__id_product_attribute, UnitProducts.price_incl_per_unit AS UnitProducts__price_incl_per_unit, UnitProducts.purchase_price_incl_per_unit AS UnitProducts__purchase_price_incl_per_unit, UnitProducts.name AS UnitProducts__name, UnitProducts.amount AS UnitProducts__amount, UnitProducts.price_per_unit_enabled AS UnitProducts__price_per_unit_enabled, UnitProducts.quantity_in_units AS UnitProducts__quantity_in_units, StockAvailables.id_stock_available AS StockAvailables__id_stock_available, StockAvailables.id_product AS StockAvailables__id_product, StockAvailables.id_product_attribute AS StockAvailables__id_product_attribute, StockAvailables.quantity AS StockAvailables__quantity, StockAvailables.quantity_limit AS StockAvailables__quantity_limit, StockAvailables.sold_out_limit AS StockAvailables__sold_out_limit, StockAvailables.always_available AS StockAvailables__always_available, StockAvailables.default_quantity_after_sending_order_lists AS StockAvailables__default_quantity_after_sending_order_lists FROM fcs_product Products INNER JOIN fcs_category_product CategoryProducts ON (CategoryProducts.id_category = 20 AND Products.id_product = CategoryProducts.id_product) LEFT JOIN fcs_images Images ON Products.id_product = Images.id_product LEFT JOIN fcs_deposits DepositProducts ON Products.id_product = DepositProducts.id_product LEFT JOIN fcs_manufacturer Manufacturers ON Manufacturers.id_manufacturer = Products.id_manufacturer LEFT JOIN fcs_stock_available StockAvailables ON (StockAvailables.id_product_attribute = 0 AND Products.id_product = StockAvailables.id_product) LEFT JOIN fcs_units UnitProducts ON Products.id_product = UnitProducts.id_product LEFT JOIN fcs_tax Taxes ON Taxes.id_tax = Products.id_tax WHERE (Products.active = 1 AND Manufacturers.active = TRUE AND Products.new &gt; '2025-05-27') ORDER BY Products.name ASC, Images.id_image DESC</p><div class="error-suggestion" style="box-sizing: border-box; margin-bottom: 20px;"><p class="notice" style="box-sizing: border-box;">If you are using SQL keywords as table column names, you can enable identifier quoting for your database connection in config/app.php.</p><p class="notice" style="box-sizing: border-box;"><strong style="box-sizing: border-box;">SQL Query:</strong></p><pre style="box-sizing: border-box; background: rgb(255, 255, 255); border-radius: 5px; padding: 5px; white-space: pre-wrap; margin: 0px;">SELECT DISTINCT Products.id_product AS Products__id_product, Products.id_manufacturer AS Products__id_manufacturer, Products.id_tax AS Products__id_tax, Products.id_storage_location AS Products__id_storage_location, Products.price AS Products__price, Products.name AS Products__name, Products.description AS Products__description, Products.description_short AS Products__description_short, Products.unity AS Products__unity, Products.is_declaration_ok AS Products__is_declaration_ok, Products.is_stock_product AS Products__is_stock_product, Products.active AS Products__active, Products.delivery_rhythm_type AS Products__delivery_rhythm_type, Products.delivery_rhythm_count AS Products__delivery_rhythm_count, Products.delivery_rhythm_first_delivery_day AS Products__delivery_rhythm_first_delivery_day, Products.delivery_rhythm_order_possible_until AS Products__delivery_rhythm_order_possible_until, Products.delivery_rhythm_send_order_list_weekday AS Products__delivery_rhythm_send_order_list_weekday, Products.delivery_rhythm_send_order_list_day AS Products__delivery_rhythm_send_order_list_day, Products.created AS Products__created, Products.modified AS Products__modified, DepositProducts.id AS DepositProducts__id, DepositProducts.id_product AS DepositProducts__id_product, DepositProducts.id_product_attribute AS DepositProducts__id_product_attribute, DepositProducts.deposit AS DepositProducts__deposit, Images.id_image AS Images__id_image, Taxes.id_tax AS Taxes__id_tax, Taxes.rate AS Taxes__rate, Taxes.active AS Taxes__active, Taxes.deleted AS Taxes__deleted, Manufacturers.id_manufacturer AS Manufacturers__id_manufacturer, Manufacturers.name AS Manufacturers__name, Manufacturers.description AS Manufacturers__description, Manufacturers.short_description AS Manufacturers__short_description, Manufacturers.created AS Manufacturers__created, Manufacturers.modified AS Manufacturers__modified, Manufacturers.active AS Manufacturers__active, Manufacturers.is_private AS Manufacturers__is_private, Manufacturers.uid_number AS Manufacturers__uid_number, Manufacturers.additional_text_for_invoice AS Manufacturers__additional_text_for_invoice, Manufacturers.iban AS Manufacturers__iban, Manufacturers.bic AS Manufacturers__bic, Manufacturers.bank_name AS Manufacturers__bank_name, Manufacturers.firmenbuchnummer AS Manufacturers__firmenbuchnummer, Manufacturers.firmengericht AS Manufacturers__firmengericht, Manufacturers.aufsichtsbehoerde AS Manufacturers__aufsichtsbehoerde, Manufacturers.kammer AS Manufacturers__kammer, Manufacturers.homepage AS Manufacturers__homepage, Manufacturers.id_customer AS Manufacturers__id_customer, Manufacturers.variable_member_fee AS Manufacturers__variable_member_fee, Manufacturers.send_invoice AS Manufacturers__send_invoice, Manufacturers.send_order_list AS Manufacturers__send_order_list, Manufacturers.default_tax_id AS Manufacturers__default_tax_id, Manufacturers.default_tax_id_purchase_price AS Manufacturers__default_tax_id_purchase_price, Manufacturers.send_order_list_cc AS Manufacturers__send_order_list_cc, Manufacturers.send_instant_order_notification AS Manufacturers__send_instant_order_notification, Manufacturers.send_ordered_product_deleted_notification AS Manufacturers__send_ordered_product_deleted_notification, Manufacturers.send_ordered_product_price_changed_notification AS Manufacturers__send_ordered_product_price_changed_notification, Manufacturers.send_ordered_product_amount_changed_notification AS Manufacturers__send_ordered_product_amount_changed_notification, Manufacturers.enabled_sync_domains AS Manufacturers__enabled_sync_domains, Manufacturers.stock_management_enabled AS Manufacturers__stock_management_enabled, Manufacturers.send_product_sold_out_limit_reached_for_manufacturer AS Manufacturers__send_product_sold_out_limit_reached_for_manufacturer, Manufacturers.send_product_sold_out_limit_reached_for_contact_person AS Manufacturers__send_product_sold_out_limit_reached_for_contact_person, Manufacturers.no_delivery_days AS Manufacturers__no_delivery_days, Manufacturers.include_stock_products_in_order_lists AS Manufacturers__include_stock_products_in_order_lists, Manufacturers.send_delivery_notes AS Manufacturers__send_delivery_notes, Manufacturers.anonymize_customers AS Manufacturers__anonymize_customers, UnitProducts.id AS UnitProducts__id, UnitProducts.id_product AS UnitProducts__id_product, UnitProducts.id_product_attribute AS UnitProducts__id_product_attribute, UnitProducts.price_incl_per_unit AS UnitProducts__price_incl_per_unit, UnitProducts.purchase_price_incl_per_unit AS UnitProducts__purchase_price_incl_per_unit, UnitProducts.name AS UnitProducts__name, UnitProducts.amount AS UnitProducts__amount, UnitProducts.price_per_unit_enabled AS UnitProducts__price_per_unit_enabled, UnitProducts.quantity_in_units AS UnitProducts__quantity_in_units, StockAvailables.id_stock_available AS StockAvailables__id_stock_available, StockAvailables.id_product AS StockAvailables__id_product, StockAvailables.id_product_attribute AS StockAvailables__id_product_attribute, StockAvailables.quantity AS StockAvailables__quantity, StockAvailables.quantity_limit AS StockAvailables__quantity_limit, StockAvailables.sold_out_limit AS StockAvailables__sold_out_limit, StockAvailables.always_available AS StockAvailables__always_available, StockAvailables.default_quantity_after_sending_order_lists AS StockAvailables__default_quantity_after_sending_order_lists FROM fcs_product Products INNER JOIN fcs_category_product CategoryProducts ON (CategoryProducts.id_category = 20 AND Products.id_product = CategoryProducts.id_product) LEFT JOIN fcs_images Images ON Products.id_product = Images.id_product LEFT JOIN fcs_deposits DepositProducts ON Products.id_product = DepositProducts.id_product LEFT JOIN fcs_manufacturer Manufacturers ON Manufacturers.id_manufacturer = Products.id_manufacturer LEFT JOIN fcs_stock_available StockAvailables ON (StockAvailables.id_product_attribute = 0 AND Products.id_product = StockAvailables.id_product) LEFT JOIN fcs_units UnitProducts ON Products.id_product = UnitProducts.id_product LEFT JOIN fcs_tax Taxes ON Taxes.id_tax = Products.id_tax WHERE (Products.active = 1 AND Manufacturers.active = TRUE AND Products.new &gt; '2025-05-27') ORDER BY Products.name ASC, Images.id_image DESC</pre></div><div class="stack-frame" style="box-sizing: border-box; background: rgb(234, 234, 234); padding: 10px; border-bottom: 2px solid rgb(245, 247, 250); overflow: hidden;"><span class="stack-frame-file" style="box-sizing: border-box; font-family: consolas, monospace; word-break: break-all;">CORE/src/Database/Driver.php at line 383<span> </span></span><a href="phpstorm://open?file=/app/vendor/cakephp/cakephp/src/Database/Driver.php&amp;line=383" class="stack-frame-edit" style="box-sizing: border-box; margin: 0px 5px 0px 0px; color: rgb(33, 33, 33); text-decoration: none;">(edit)</a>
  | 'driver' => $this,
-- | --
  | 'params' => $params ?? $statement->getBoundParams(),
  | ]);
  |  
  | return new QueryException($loggedQuery, $exception);
  | }
  |  
  | /**
  | * Prepares a sql statement to be executed.

</div><ul class="stack-frames" style="box-sizing: border-box; list-style: none; padding: 0px; margin: 0px; border-radius: 5px;"><li id="stack-frame-1-0" class="stack-frame vendor-frame" style="box-sizing: border-box; background: rgb(241, 241, 241); padding: 10px; border-bottom: 2px solid rgb(245, 247, 250); overflow: hidden; border-radius: 5px 5px 0px 0px;"><div class="stack-frame-header" style="box-sizing: border-box; display: flex; align-items: center; gap: 10px;"><button data-frame-id="1-0" class="stack-frame-toggle" style="box-sizing: border-box; cursor: pointer; color: rgb(82, 82, 82); border: 1px solid rgb(210, 210, 210); border-radius: 5px; height: 28px; width: 28px; background: rgb(245, 247, 250); line-height: 1.5; opacity: 0.7;">▼</button><div class="stack-frame-header-content" style="box-sizing: border-box; display: flex; gap: 8px; color: rgb(124, 124, 124);"><span class="stack-frame-file" style="box-sizing: border-box; font-family: consolas, monospace; word-break: break-all;">CORE/src/Database/Statement/Statement.php</span><span class="stack-frame-line" style="box-sizing: border-box; font-family: consolas, monospace;"><span class="stack-frame-label" style="box-sizing: border-box; font-family: &quot;Helvetica Neue&quot;, Helvetica, Arial, sans-serif; font-weight: normal; margin: 0px 5px 0px 0px; font-size: 0.9em;">at line</span>144</span><span class="stack-function" style="box-sizing: border-box; font-family: consolas, monospace;"><span class="stack-frame-label" style="box-sizing: border-box; font-family: &quot;Helvetica Neue&quot;, Helvetica, Arial, sans-serif; font-weight: normal; margin: 0px; font-size: 0.9em;">in</span><span> </span>PDOStatement-&gt;execute</span><a href="phpstorm://open?file=/app/vendor/cakephp/cakephp/src/Database/Statement/Statement.php&amp;line=144" class="stack-frame-edit" style="box-sizing: border-box; margin: 0px 5px 0px 0px; color: rgb(124, 124, 124); text-decoration: none;">(edit)</a></div></div></li><li id="stack-frame-1-1" class="stack-frame vendor-frame" style="box-sizing: border-box; background: rgb(241, 241, 241); padding: 10px; border-bottom: 2px solid rgb(245, 247, 250); overflow: hidden;"><div class="stack-frame-header" style="box-sizing: border-box; display: flex; align-items: center; gap: 10px;"><button data-frame-id="1-1" class="stack-frame-toggle" style="box-sizing: border-box; cursor: pointer; color: rgb(82, 82, 82); border: 1px solid rgb(210, 210, 210); border-radius: 5px; height: 28px; width: 28px; background: rgb(245, 247, 250); line-height: 1.5; opacity: 0.7;">▼</button><div class="stack-frame-header-content" style="box-sizing: border-box; display: flex; gap: 8px; color: rgb(124, 124, 124);"><span class="stack-frame-file" style="box-sizing: border-box; font-family: consolas, monospace; word-break: break-all;">CORE/src/Database/Driver.php</span><span class="stack-frame-line" style="box-sizing: border-box; font-family: consolas, monospace;"><span class="stack-frame-label" style="box-sizing: border-box; font-family: &quot;Helvetica Neue&quot;, Helvetica, Arial, sans-serif; font-weight: normal; margin: 0px 5px 0px 0px; font-size: 0.9em;">at line</span>341</span><span class="stack-function" style="box-sizing: border-box; font-family: consolas, monospace;"><span class="stack-frame-label" style="box-sizing: border-box; font-family: &quot;Helvetica Neue&quot;, Helvetica, Arial, sans-serif; font-weight: normal; margin: 0px; font-size: 0.9em;">in</span><span> </span>Cake\Database\Statement\Statement-&gt;execute</span><a href="phpstorm://open?file=/app/vendor/cakephp/cakephp/src/Database/Driver.php&amp;line=341" class="stack-frame-edit" style="box-sizing: border-box; margin: 0px 5px 0px 0px; color: rgb(124, 124, 124); text-decoration: none;">(edit)</a></div></div></li><li id="stack-frame-1-2" class="stack-frame " style="box-sizing: border-box; background: rgb(234, 234, 234); padding: 10px; border-bottom: none; overflow: hidden; border-radius: 0px 0px 5px 5px; margin-bottom: 0px;"><div class="stack-frame-header" style="box-sizing: border-box; display: flex; align-items: center; gap: 10px;"><button data-frame-id="1-2" class="stack-frame-toggle" style="box-sizing: border-box; cursor: pointer; color: rgb(82, 82, 82); border: 1px solid rgb(210, 210, 210); border-radius: 5px; height: 28px; width: 28px; background: rgb(245, 247, 250); line-height: 1.5; opacity: 0.7;">▼</button><div class="stack-frame-header-content" style="box-sizing: border-box; display: flex; gap: 8px;"><span class="stack-frame-file" style="box-sizing: border-box; font-family: consolas, monospace; word-break: break-all;">[main]</span><span class="stack-function" style="box-sizing: border-box; font-family: consolas, monospace;"></span></div></div></li></ul><p class="customize" style="box-sizing: border-box;">If you want to customize this error message, create<span> </span><em style="box-sizing: border-box;">templates/Error/pdo_error.php</em></p></div><iframe height="40" width="40" src="http://localhost/debug-kit/toolbar/03cd9345-b46e-4dd6-b7e7-d57b3c0732d4" style="box-sizing: border-box; color: rgb(64, 64, 65); font-family: &quot;Helvetica Neue&quot;, Helvetica, Arial, sans-serif; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: 0.14px; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial; position: fixed; bottom: 0px; right: 0px; border: 0px; outline: 0px; overflow: hidden; z-index: 99999;"></iframe><!--EndFragment-->
</body>
</html>Database Error
Cake\Database\Exception\QueryException
Error SQLSTATE[42S22]: Column not found: 1054 Unknown column 'Products.new' in 'where clause' Query: SELECT DISTINCT Products.id_product AS Products__id_product, Products.id_manufacturer AS Products__id_manufacturer, Products.id_tax AS Products__id_tax, Products.id_storage_location AS Products__id_storage_location, Products.price AS Products__price, Products.name AS Products__name, Products.description AS Products__description, Products.description_short AS Products__description_short, Products.unity AS Products__unity, Products.is_declaration_ok AS Products__is_declaration_ok, Products.is_stock_product AS Products__is_stock_product, Products.active AS Products__active, Products.delivery_rhythm_type AS Products__delivery_rhythm_type, Products.delivery_rhythm_count AS Products__delivery_rhythm_count, Products.delivery_rhythm_first_delivery_day AS Products__delivery_rhythm_first_delivery_day, Products.delivery_rhythm_order_possible_until AS Products__delivery_rhythm_order_possible_until, Products.delivery_rhythm_send_order_list_weekday AS Products__delivery_rhythm_send_order_list_weekday, Products.delivery_rhythm_send_order_list_day AS Products__delivery_rhythm_send_order_list_day, Products.created AS Products__created, Products.modified AS Products__modified, DepositProducts.id AS DepositProducts__id, DepositProducts.id_product AS DepositProducts__id_product, DepositProducts.id_product_attribute AS DepositProducts__id_product_attribute, DepositProducts.deposit AS DepositProducts__deposit, Images.id_image AS Images__id_image, Taxes.id_tax AS Taxes__id_tax, Taxes.rate AS Taxes__rate, Taxes.active AS Taxes__active, Taxes.deleted AS Taxes__deleted, Manufacturers.id_manufacturer AS Manufacturers__id_manufacturer, Manufacturers.name AS Manufacturers__name, Manufacturers.description AS Manufacturers__description, Manufacturers.short_description AS Manufacturers__short_description, Manufacturers.created AS Manufacturers__created, Manufacturers.modified AS Manufacturers__modified, Manufacturers.active AS Manufacturers__active, Manufacturers.is_private AS Manufacturers__is_private, Manufacturers.uid_number AS Manufacturers__uid_number, Manufacturers.additional_text_for_invoice AS Manufacturers__additional_text_for_invoice, Manufacturers.iban AS Manufacturers__iban, Manufacturers.bic AS Manufacturers__bic, Manufacturers.bank_name AS Manufacturers__bank_name, Manufacturers.firmenbuchnummer AS Manufacturers__firmenbuchnummer, Manufacturers.firmengericht AS Manufacturers__firmengericht, Manufacturers.aufsichtsbehoerde AS Manufacturers__aufsichtsbehoerde, Manufacturers.kammer AS Manufacturers__kammer, Manufacturers.homepage AS Manufacturers__homepage, Manufacturers.id_customer AS Manufacturers__id_customer, Manufacturers.variable_member_fee AS Manufacturers__variable_member_fee, Manufacturers.send_invoice AS Manufacturers__send_invoice, Manufacturers.send_order_list AS Manufacturers__send_order_list, Manufacturers.default_tax_id AS Manufacturers__default_tax_id, Manufacturers.default_tax_id_purchase_price AS Manufacturers__default_tax_id_purchase_price, Manufacturers.send_order_list_cc AS Manufacturers__send_order_list_cc, Manufacturers.send_instant_order_notification AS Manufacturers__send_instant_order_notification, Manufacturers.send_ordered_product_deleted_notification AS Manufacturers__send_ordered_product_deleted_notification, Manufacturers.send_ordered_product_price_changed_notification AS Manufacturers__send_ordered_product_price_changed_notification, Manufacturers.send_ordered_product_amount_changed_notification AS Manufacturers__send_ordered_product_amount_changed_notification, Manufacturers.enabled_sync_domains AS Manufacturers__enabled_sync_domains, Manufacturers.stock_management_enabled AS Manufacturers__stock_management_enabled, Manufacturers.send_product_sold_out_limit_reached_for_manufacturer AS Manufacturers__send_product_sold_out_limit_reached_for_manufacturer, Manufacturers.send_product_sold_out_limit_reached_for_contact_person AS Manufacturers__send_product_sold_out_limit_reached_for_contact_person, Manufacturers.no_delivery_days AS Manufacturers__no_delivery_days, Manufacturers.include_stock_products_in_order_lists AS Manufacturers__include_stock_products_in_order_lists, Manufacturers.send_delivery_notes AS Manufacturers__send_delivery_notes, Manufacturers.anonymize_customers AS Manufacturers__anonymize_customers, UnitProducts.id AS UnitProducts__id, UnitProducts.id_product AS UnitProducts__id_product, UnitProducts.id_product_attribute AS UnitProducts__id_product_attribute, UnitProducts.price_incl_per_unit AS UnitProducts__price_incl_per_unit, UnitProducts.purchase_price_incl_per_unit AS UnitProducts__purchase_price_incl_per_unit, UnitProducts.name AS UnitProducts__name, UnitProducts.amount AS UnitProducts__amount, UnitProducts.price_per_unit_enabled AS UnitProducts__price_per_unit_enabled, UnitProducts.quantity_in_units AS UnitProducts__quantity_in_units, StockAvailables.id_stock_available AS StockAvailables__id_stock_available, StockAvailables.id_product AS StockAvailables__id_product, StockAvailables.id_product_attribute AS StockAvailables__id_product_attribute, StockAvailables.quantity AS StockAvailables__quantity, StockAvailables.quantity_limit AS StockAvailables__quantity_limit, StockAvailables.sold_out_limit AS StockAvailables__sold_out_limit, StockAvailables.always_available AS StockAvailables__always_available, StockAvailables.default_quantity_after_sending_order_lists AS StockAvailables__default_quantity_after_sending_order_lists FROM fcs_product Products INNER JOIN fcs_category_product CategoryProducts ON (CategoryProducts.id_category = 20 AND Products.id_product = CategoryProducts.id_product) LEFT JOIN fcs_images Images ON Products.id_product = Images.id_product LEFT JOIN fcs_deposits DepositProducts ON Products.id_product = DepositProducts.id_product LEFT JOIN fcs_manufacturer Manufacturers ON Manufacturers.id_manufacturer = Products.id_manufacturer LEFT JOIN fcs_stock_available StockAvailables ON (StockAvailables.id_product_attribute = 0 AND Products.id_product = StockAvailables.id_product) LEFT JOIN fcs_units UnitProducts ON Products.id_product = UnitProducts.id_product LEFT JOIN fcs_tax Taxes ON Taxes.id_tax = Products.id_tax WHERE (Products.active = 1 AND Manufacturers.active = TRUE AND Products.new > '2025-05-27') ORDER BY Products.name ASC, Images.id_image DESC

If you are using SQL keywords as table column names, you can enable identifier quoting for your database connection in config/app.php.

SQL Query:

SELECT DISTINCT Products.id_product AS Products__id_product, Products.id_manufacturer AS Products__id_manufacturer, Products.id_tax AS Products__id_tax, Products.id_storage_location AS Products__id_storage_location, Products.price AS Products__price, Products.name AS Products__name, Products.description AS Products__description, Products.description_short AS Products__description_short, Products.unity AS Products__unity, Products.is_declaration_ok AS Products__is_declaration_ok, Products.is_stock_product AS Products__is_stock_product, Products.active AS Products__active, Products.delivery_rhythm_type AS Products__delivery_rhythm_type, Products.delivery_rhythm_count AS Products__delivery_rhythm_count, Products.delivery_rhythm_first_delivery_day AS Products__delivery_rhythm_first_delivery_day, Products.delivery_rhythm_order_possible_until AS Products__delivery_rhythm_order_possible_until, Products.delivery_rhythm_send_order_list_weekday AS Products__delivery_rhythm_send_order_list_weekday, Products.delivery_rhythm_send_order_list_day AS Products__delivery_rhythm_send_order_list_day, Products.created AS Products__created, Products.modified AS Products__modified, DepositProducts.id AS DepositProducts__id, DepositProducts.id_product AS DepositProducts__id_product, DepositProducts.id_product_attribute AS DepositProducts__id_product_attribute, DepositProducts.deposit AS DepositProducts__deposit, Images.id_image AS Images__id_image, Taxes.id_tax AS Taxes__id_tax, Taxes.rate AS Taxes__rate, Taxes.active AS Taxes__active, Taxes.deleted AS Taxes__deleted, Manufacturers.id_manufacturer AS Manufacturers__id_manufacturer, Manufacturers.name AS Manufacturers__name, Manufacturers.description AS Manufacturers__description, Manufacturers.short_description AS Manufacturers__short_description, Manufacturers.created AS Manufacturers__created, Manufacturers.modified AS Manufacturers__modified, Manufacturers.active AS Manufacturers__active, Manufacturers.is_private AS Manufacturers__is_private, Manufacturers.uid_number AS Manufacturers__uid_number, Manufacturers.additional_text_for_invoice AS Manufacturers__additional_text_for_invoice, Manufacturers.iban AS Manufacturers__iban, Manufacturers.bic AS Manufacturers__bic, Manufacturers.bank_name AS Manufacturers__bank_name, Manufacturers.firmenbuchnummer AS Manufacturers__firmenbuchnummer, Manufacturers.firmengericht AS Manufacturers__firmengericht, Manufacturers.aufsichtsbehoerde AS Manufacturers__aufsichtsbehoerde, Manufacturers.kammer AS Manufacturers__kammer, Manufacturers.homepage AS Manufacturers__homepage, Manufacturers.id_customer AS Manufacturers__id_customer, Manufacturers.variable_member_fee AS Manufacturers__variable_member_fee, Manufacturers.send_invoice AS Manufacturers__send_invoice, Manufacturers.send_order_list AS Manufacturers__send_order_list, Manufacturers.default_tax_id AS Manufacturers__default_tax_id, Manufacturers.default_tax_id_purchase_price AS Manufacturers__default_tax_id_purchase_price, Manufacturers.send_order_list_cc AS Manufacturers__send_order_list_cc, Manufacturers.send_instant_order_notification AS Manufacturers__send_instant_order_notification, Manufacturers.send_ordered_product_deleted_notification AS Manufacturers__send_ordered_product_deleted_notification, Manufacturers.send_ordered_product_price_changed_notification AS Manufacturers__send_ordered_product_price_changed_notification, Manufacturers.send_ordered_product_amount_changed_notification AS Manufacturers__send_ordered_product_amount_changed_notification, Manufacturers.enabled_sync_domains AS Manufacturers__enabled_sync_domains, Manufacturers.stock_management_enabled AS Manufacturers__stock_management_enabled, Manufacturers.send_product_sold_out_limit_reached_for_manufacturer AS Manufacturers__send_product_sold_out_limit_reached_for_manufacturer, Manufacturers.send_product_sold_out_limit_reached_for_contact_person AS Manufacturers__send_product_sold_out_limit_reached_for_contact_person, Manufacturers.no_delivery_days AS Manufacturers__no_delivery_days, Manufacturers.include_stock_products_in_order_lists AS Manufacturers__include_stock_products_in_order_lists, Manufacturers.send_delivery_notes AS Manufacturers__send_delivery_notes, Manufacturers.anonymize_customers AS Manufacturers__anonymize_customers, UnitProducts.id AS UnitProducts__id, UnitProducts.id_product AS UnitProducts__id_product, UnitProducts.id_product_attribute AS UnitProducts__id_product_attribute, UnitProducts.price_incl_per_unit AS UnitProducts__price_incl_per_unit, UnitProducts.purchase_price_incl_per_unit AS UnitProducts__purchase_price_incl_per_unit, UnitProducts.name AS UnitProducts__name, UnitProducts.amount AS UnitProducts__amount, UnitProducts.price_per_unit_enabled AS UnitProducts__price_per_unit_enabled, UnitProducts.quantity_in_units AS UnitProducts__quantity_in_units, StockAvailables.id_stock_available AS StockAvailables__id_stock_available, StockAvailables.id_product AS StockAvailables__id_product, StockAvailables.id_product_attribute AS StockAvailables__id_product_attribute, StockAvailables.quantity AS StockAvailables__quantity, StockAvailables.quantity_limit AS StockAvailables__quantity_limit, StockAvailables.sold_out_limit AS StockAvailables__sold_out_limit, StockAvailables.always_available AS StockAvailables__always_available, StockAvailables.default_quantity_after_sending_order_lists AS StockAvailables__default_quantity_after_sending_order_lists FROM fcs_product Products INNER JOIN fcs_category_product CategoryProducts ON (CategoryProducts.id_category = 20 AND Products.id_product = CategoryProducts.id_product) LEFT JOIN fcs_images Images ON Products.id_product = Images.id_product LEFT JOIN fcs_deposits DepositProducts ON Products.id_product = DepositProducts.id_product LEFT JOIN fcs_manufacturer Manufacturers ON Manufacturers.id_manufacturer = Products.id_manufacturer LEFT JOIN fcs_stock_available StockAvailables ON (StockAvailables.id_product_attribute = 0 AND Products.id_product = StockAvailables.id_product) LEFT JOIN fcs_units UnitProducts ON Products.id_product = UnitProducts.id_product LEFT JOIN fcs_tax Taxes ON Taxes.id_tax = Products.id_tax WHERE (Products.active = 1 AND Manufacturers.active = TRUE AND Products.new > '2025-05-27') ORDER BY Products.name ASC, Images.id_image DESC
CORE/src/Database/Driver.php at line 383 [(edit)](phpstorm://open?file=/app/vendor/cakephp/cakephp/src/Database/Driver.php&line=383)
            'driver' => $this,
            'params' => $params ?? $statement->getBoundParams(),
        ]);
        return new QueryException($loggedQuery, $exception);
    }
    /**
     * Prepares a sql statement to be executed.
▼
CORE/src/Database/Driver.php
at line359
in Cake\Database\Driver->createQueryException
[(edit)](phpstorm://open?file=/app/vendor/cakephp/cakephp/src/Database/Driver.php&line=359)
▼
CORE/src/Database/Driver/Mysql.php
at line178
in Cake\Database\Driver->executeStatement
[(edit)](phpstorm://open?file=/app/vendor/cakephp/cakephp/src/Database/Driver/Mysql.php&line=178)
▼
CORE/src/Database/Connection.php
at line280
in Cake\Database\Driver\Mysql->run
[(edit)](phpstorm://open?file=/app/vendor/cakephp/cakephp/src/Database/Connection.php&line=280)
▼
CORE/src/Core/Retry/CommandRetry.php
at line71
in Cake\Database\Connection->Cake\Database\{closure}
[(edit)](phpstorm://open?file=/app/vendor/cakephp/cakephp/src/Core/Retry/CommandRetry.php&line=71)
▼
CORE/src/Database/Connection.php
at line280
in Cake\Core\Retry\CommandRetry->run
[(edit)](phpstorm://open?file=/app/vendor/cakephp/cakephp/src/Database/Connection.php&line=280)
▼
CORE/src/Database/Query.php
at line224
in Cake\Database\Connection->run
[(edit)](phpstorm://open?file=/app/vendor/cakephp/cakephp/src/Database/Query.php&line=224)
▼
CORE/src/Database/Query/SelectQuery.php
at line123
in Cake\Database\Query->execute
[(edit)](phpstorm://open?file=/app/vendor/cakephp/cakephp/src/Database/Query/SelectQuery.php&line=123)
▼
CORE/src/ORM/Query/SelectQuery.php
at line1570
in Cake\Database\Query\SelectQuery->all
[(edit)](phpstorm://open?file=/app/vendor/cakephp/cakephp/src/ORM/Query/SelectQuery.php&line=1570)
▼
CORE/src/ORM/Query/SelectQuery.php
at line382
in Cake\ORM\Query\SelectQuery->_execute
[(edit)](phpstorm://open?file=/app/vendor/cakephp/cakephp/src/ORM/Query/SelectQuery.php&line=382)
▼
CORE/src/ORM/Query/SelectQuery.php
at line397
in Cake\ORM\Query\SelectQuery->all
[(edit)](phpstorm://open?file=/app/vendor/cakephp/cakephp/src/ORM/Query/SelectQuery.php&line=397)
▼
APP/Services/CatalogService.php
at line94
in Cake\ORM\Query\SelectQuery->toArray
[(edit)](phpstorm://open?file=/app/src/Services/CatalogService.php&line=94)
▼
APP/Controller/PagesController.php
at line57
in App\Services\CatalogService->getProducts
[(edit)](phpstorm://open?file=/app/src/Controller/PagesController.php&line=57)
▼
CORE/src/Controller/Controller.php
at line505
in App\Controller\PagesController->home
[(edit)](phpstorm://open?file=/app/vendor/cakephp/cakephp/src/Controller/Controller.php&line=505)
▼
CORE/src/Controller/ControllerFactory.php
at line166
in Cake\Controller\Controller->invokeAction
[(edit)](phpstorm://open?file=/app/vendor/cakephp/cakephp/src/Controller/ControllerFactory.php&line=166)
▼
CORE/src/Controller/ControllerFactory.php
at line141
in Cake\Controller\ControllerFactory->handle
[(edit)](phpstorm://open?file=/app/vendor/cakephp/cakephp/src/Controller/ControllerFactory.php&line=141)
▼
CORE/src/Http/BaseApplication.php
at line362
in Cake\Controller\ControllerFactory->invoke
[(edit)](phpstorm://open?file=/app/vendor/cakephp/cakephp/src/Http/BaseApplication.php&line=362)
▼
CORE/src/Http/Runner.php
at line86
in Cake\Http\BaseApplication->handle
[(edit)](phpstorm://open?file=/app/vendor/cakephp/cakephp/src/Http/Runner.php&line=86)
▼
ROOT/vendor/markstory/asset_compress/src/Middleware/AssetCompressMiddleware.php
at line64
in Cake\Http\Runner->handle
[(edit)](phpstorm://open?file=/app/vendor/markstory/asset_compress/src/Middleware/AssetCompressMiddleware.php&line=64)
▼
CORE/src/Http/Runner.php
at line82
in AssetCompress\Middleware\AssetCompressMiddleware->process
[(edit)](phpstorm://open?file=/app/vendor/cakephp/cakephp/src/Http/Runner.php&line=82)
▼
CORE/src/Error/Middleware/ErrorHandlerMiddleware.php
at line115
in Cake\Http\Runner->handle
[(edit)](phpstorm://open?file=/app/vendor/cakephp/cakephp/src/Error/Middleware/ErrorHandlerMiddleware.php&line=115)
▼
CORE/src/Http/Runner.php
at line82
in Cake\Error\Middleware\ErrorHandlerMiddleware->process
[(edit)](phpstorm://open?file=/app/vendor/cakephp/cakephp/src/Http/Runner.php&line=82)
▼
ROOT/vendor/cakephp/authorization/src/Middleware/RequestAuthorizationMiddleware.php
at line110
in Cake\Http\Runner->handle
[(edit)](phpstorm://open?file=/app/vendor/cakephp/authorization/src/Middleware/RequestAuthorizationMiddleware.php&line=110)
▼
CORE/src/Http/Runner.php
at line82
in Authorization\Middleware\RequestAuthorizationMiddleware->process
[(edit)](phpstorm://open?file=/app/vendor/cakephp/cakephp/src/Http/Runner.php&line=82)
▼
ROOT/vendor/cakephp/authorization/src/Middleware/AuthorizationMiddleware.php
at line136
in Cake\Http\Runner->handle
[(edit)](phpstorm://open?file=/app/vendor/cakephp/authorization/src/Middleware/AuthorizationMiddleware.php&line=136)
▼
CORE/src/Http/Runner.php
at line82
in Authorization\Middleware\AuthorizationMiddleware->process
[(edit)](phpstorm://open?file=/app/vendor/cakephp/cakephp/src/Http/Runner.php&line=82)
▼
ROOT/vendor/cakephp/authentication/src/Middleware/AuthenticationMiddleware.php
at line106
in Cake\Http\Runner->handle
[(edit)](phpstorm://open?file=/app/vendor/cakephp/authentication/src/Middleware/AuthenticationMiddleware.php&line=106)
▼
CORE/src/Http/Runner.php
at line82
in Authentication\Middleware\AuthenticationMiddleware->process
[(edit)](phpstorm://open?file=/app/vendor/cakephp/cakephp/src/Http/Runner.php&line=82)
▼
CORE/src/Http/Middleware/EncryptedCookieMiddleware.php
at line93
in Cake\Http\Runner->handle
[(edit)](phpstorm://open?file=/app/vendor/cakephp/cakephp/src/Http/Middleware/EncryptedCookieMiddleware.php&line=93)
▼
CORE/src/Http/Runner.php
at line82
in Cake\Http\Middleware\EncryptedCookieMiddleware->process
[(edit)](phpstorm://open?file=/app/vendor/cakephp/cakephp/src/Http/Runner.php&line=82)
▼
CORE/src/Routing/Middleware/RoutingMiddleware.php
at line117
in Cake\Http\Runner->handle
[(edit)](phpstorm://open?file=/app/vendor/cakephp/cakephp/src/Routing/Middleware/RoutingMiddleware.php&line=117)
▼
CORE/src/Http/Runner.php
at line82
in Cake\Routing\Middleware\RoutingMiddleware->process
[(edit)](phpstorm://open?file=/app/vendor/cakephp/cakephp/src/Http/Runner.php&line=82)
▼
CORE/src/Http/Middleware/CsrfProtectionMiddleware.php
at line169
in Cake\Http\Runner->handle
[(edit)](phpstorm://open?file=/app/vendor/cakephp/cakephp/src/Http/Middleware/CsrfProtectionMiddleware.php&line=169)
▼
CORE/src/Http/Runner.php
at line82
in Cake\Http\Middleware\CsrfProtectionMiddleware->process
[(edit)](phpstorm://open?file=/app/vendor/cakephp/cakephp/src/Http/Runner.php&line=82)
▼
CORE/src/Routing/Middleware/AssetMiddleware.php
at line70
in Cake\Http\Runner->handle
[(edit)](phpstorm://open?file=/app/vendor/cakephp/cakephp/src/Routing/Middleware/AssetMiddleware.php&line=70)
▼
CORE/src/Http/Runner.php
at line82
in Cake\Routing\Middleware\AssetMiddleware->process
[(edit)](phpstorm://open?file=/app/vendor/cakephp/cakephp/src/Http/Runner.php&line=82)
▼
ROOT/vendor/cakephp/debug_kit/src/Middleware/DebugKitMiddleware.php
at line60
in Cake\Http\Runner->handle
[(edit)](phpstorm://open?file=/app/vendor/cakephp/debug_kit/src/Middleware/DebugKitMiddleware.php&line=60)
▼
CORE/src/Http/Runner.php
at line82
in DebugKit\Middleware\DebugKitMiddleware->process
[(edit)](phpstorm://open?file=/app/vendor/cakephp/cakephp/src/Http/Runner.php&line=82)
▼
CORE/src/Http/Runner.php
at line60
in Cake\Http\Runner->handle
[(edit)](phpstorm://open?file=/app/vendor/cakephp/cakephp/src/Http/Runner.php&line=60)
▼
CORE/src/Http/Server.php
at line104
in Cake\Http\Runner->run
[(edit)](phpstorm://open?file=/app/vendor/cakephp/cakephp/src/Http/Server.php&line=104)
▼
ROOT/webroot/index.php
at line37
in Cake\Http\Server->run
[(edit)](phpstorm://open?file=/app/webroot/index.php&line=37)
▼
[main]
Caused by
SQLSTATE[42S22]: Column not found: 1054 Unknown column 'Products.new' in 'where clause'
PDOException
CORE/src/Database/Statement/Statement.php at line 144 [(edit)](phpstorm://open?file=/app/vendor/cakephp/cakephp/src/Database/Statement/Statement.php&line=144)
     * @inheritDoc
     */
    public function execute(?array $params = null): bool
    {
        return $this->statement->execute($params);
    }
    /**
     * @inheritDoc
▼
CORE/src/Database/Statement/Statement.php
at line144
in PDOStatement->execute
[(edit)](phpstorm://open?file=/app/vendor/cakephp/cakephp/src/Database/Statement/Statement.php&line=144)
▼
CORE/src/Database/Driver.php
at line341
in Cake\Database\Statement\Statement->execute
[(edit)](phpstorm://open?file=/app/vendor/cakephp/cakephp/src/Database/Driver.php&line=341)
▼
[main]
If you want to customize this error message, create templates/Error/pdo_error.php

pabneukistl commented:
<img width="794" height="149" alt="image" src="https://github.com/user-attachments/assets/4d2948da-fdcf-4479-9f2f-65ea5879041e" />
Spalte war als "Neu" da aber ohne Name und INT... vielleicht gehört da in den Migrations noch was angepasst?

<img width="609" height="254" alt="image" src="https://github.com/user-attachments/assets/75ec6c91-6e37-42ac-baa0-25d52e27942c" />
Migrations sehen gut aus.. aber vielleicht gehört doch noch was anderes angepasst? 


hab jetzt die Spalte "new" (DATE, default NULL) manuell in der DB products eingetragen dann kann ich mal weiter testen :-)

pabneukistl commented:
so die Testerei ist so bisl mühsam, da hats generell mehr mit den Migrations? krieg bei meinem Test immer den Fehler, dass eben keine Mail gesendet wurde aber hab jetzt geschaut wenn ich mir debugge ($this->getTableLocator()->exists('Queue.QueuedJobs')); liefert das false, dh. hab die Tabelle für die QueuedJobs gar nicht?
Hätt dann probiert: 
<img width="666" height="87" alt="image" src="https://github.com/user-attachments/assets/03c654e9-6d92-4043-a496-f29be473552c" />

aber krieg hier eine permission Fehler.. wurde an den Berechtigungen vielleicht auch was gemacht?

Kannst du hier auf das Migrations-Thema mal drauf schauen?
Manueller Test bzgl. Warenkorb leeren bei SB-Benutzer und Warenkorb belassen bei anderem Benutzer so ok aber wie gesagt, kann hier aktuell leider keine Mails testen!?

mrothauer commented:
@pabneukistl merge mal develop in deinen branch, lösche alle dateien unter /vendor und mach ein frisches `composer install`, führe dann die migrations nochmal aus. da dürften sich dann die meisten probleme erledigt haben.
ps bin im urlaub

mrothauer commented on src/Controller/CartsController.php:
```diff
@@ -256,6 +275,20 @@ private function doEmptyCart(): void
+        $cartsTable = TableRegistry::getTableLocator()->get('Carts');
+        $email = new AppMailer();
+        $email->viewBuilder()->setTemplate('order_successful_self_service');
+        $email->setTo($this->identity->email)
```
hier würde sich anbieten, direkt an Configure::read('appDb.FCS_APP_EMAIL') zu versenden, ist ja für den owner interessant

mrothauer commented on tests/TestCase/Traits/QueueTrait.php:
```diff
@@ -30,7 +30,7 @@ protected function runAndAssertQueue(): void
                 pr($queuedJob->failure_message);
             }
-            $this->assertEmpty($queuedJob->failurue_message);
+            $this->assertEmpty($queuedJob->failure_message);
```
danke

mrothauer commented on webroot/js/self-service.js:
```diff
@@ -13,7 +13,7 @@
 foodcoopshop.SelfService = {
 
-    autoLogoutTimer : 180,
+    autoLogoutTimer : 5,
```
den wert hier bitte nicht einfach überschreiben, das betrifft ja sonst alle. bitte über eine neue config in app_config.php (kann dann in custom_config.php überschrieben werden) lösen
`app.selfServiceModeAutoLogoutTimer = 180`;

mrothauer commented on webroot/js/self-service.js:
```diff
@@ -261,7 +261,8 @@ foodcoopshop.SelfService = {
                 if (foodcoopshop.SelfService.currentLogoutTimer == 0) {
-                    document.location.href = '/' + foodcoopshop.LocalizedJs.helper.routeLogout + '?redirect=' + document.location.href;
+                    $redirectUrl='/' + foodcoopshop.LocalizedJs.helper.routeLogout + '?redirect=' + document.location.href;
+                    document.location.href = '/' + foodcoopshop.LocalizedJs.cart.routeCart + '/emptyCart?redirect='+$redirectUrl+'&autologout=1';
```
autologout=1 darf nur angefügt werden, wenn über buttons eingeloggt ist
weiters sind da ein bissl viel redirect-paramter vorhanden

mrothauer commented:
@pabneukistl  bin fertig mit review
